BHoM_Engine/Query/FindIdentifier.cs

```diff
@@ -37,26 +37,32 @@ public static partial class Query
         /**** Public Methods                            ****/
         /***************************************************/
 
-        [Description("Tries to find a IAdapterId on the object.")]
+        [Description("Tries to find an IAdapterId on the object.")]
+        [Input("o", "The object to search for an IAdapterId.")]
         [Output("identifier", "First plausible identifier present on the object.")]
         public static Type FindIdentifier(this IBHoMObject o)
         {
-            return o.Fragments.FirstOrDefault(fr => fr is IAdapterId)?.GetType();
+            return o?.Fragments.FirstOrDefault(fr => fr is IAdapterId)?.GetType();
```

Comment:
Some suggestions for centralising messaging

```suggestion
            if (o == null)
            {
                Engine.Reflection.Compute.RecordError("Provided object is null. Cannot extract identifier.");
                return null;
            }
            Type adapterIdType = o.Fragments.FirstOrDefault(fr => fr is IAdapterId)?.GetType();

            if (adapterIdType == null)
            {
                Engine.Reflection.Compute.RecordError("No Identifier found");
                return null;
            }
            else
            {
                Engine.Reflection.Compute.RecordNote("Auto-generated Identifier as " + adapterIdType.Name);
                return adapterIdType;
            }
```